File: src-tauri/tests/fixtures/symbol_index_languages/ruby/user_service.rb
require "json"
require_relative "user_formatter"

module Example::Users
  class UserService
    def find_user(id)
      { id: id }
    end

    def self.normalize_user_id(id)
      id.strip
    end

    def active?
      true
    end
  end
end

def format_user(user)
  JSON.generate(user)
end

File: src-tauri/tests/fixtures/symbol_index_languages/kotlin/UserService.kt
package com.example.users

import kotlinx.coroutines.Dispatchers
import com.example.shared.UserId as SharedUserId

class UserService {
    fun findUser(id: String): UserView {
        return UserView(id, "")
    }
}

interface UserFormatter {
    fun format(user: UserView): String
}

object UserCache

enum class UserStatus {
    Active,
    Disabled
}

data class UserView(val id: String, val name: String)

fun normalizeUserId(id: String): String = id.trim()
